### PR TITLE
FAD-5500 Subaccount Typeahead

### PR DESCRIPTION
**Pull request by tough-griff**
Here's an initial pass at the subaccount typeahead component that will be used in probably half of the forms in our UI.

TO DO:
- [x] Handle initial values

### CODE REVIEW

File: src/components/subaccountTypeahead/SubaccountTypeahead.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    isOpen
+  }) => {
+    const mappedMatches = subaccounts
+      .filter(
+        (item) =>
+          !inputValue ||
+          item.name.toLowerCase().includes(inputValue.toLowerCase()) ||
+          item.id.toString().includes(inputValue)
+      )
+      // .slice(0, 10) // TODO tweak this number
```
Comment (tough-griff) [inline src/components/subaccountTypeahead/SubaccountTypeahead.js]: What do you think is reasonable here?

File: src/components/subaccountTypeahead/SubaccountTypeahead.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    getItemProps,
+    getLabelProps,
+    highlightedIndex,
+    inputValue,
+    selectedItem,
+    clearSelection,
+    isOpen
+  }) => {
+    const mappedMatches = subaccounts
+      .filter(
```
Comment (tough-griff) [inline src/components/subaccountTypeahead/SubaccountTypeahead.js]: Will the `sortMatch` helper plug in here with no problems?

File: src/components/reduxFormWrappers.js [hunk 3]
```diff
@@ -1,21 +1,48 @@
...
+        id={o.label}
+        label={o.label}
+        value={o.value}
+        checked={o.value === input.value}
+      />
+    ))}
   </Radio.Group>
 );
+
+export const SubaccountTypeaheadWrapper = ({ input, subaccounts }) => (
```
Comment (tough-griff) [inline src/components/reduxFormWrappers.js]: This plugs into a redux-form `Field` like so:

```jsx
<Field
  name="subaccount"
  component={SubaccountTypeaheadWrapper}
  subaccounts={subaccountsFromReduxStore}
/>
```

Comment (jasonrhodes) [reply src/components/reduxFormWrappers.js]: Can you add this as a comment above the export here, for now? We can move it to docs eventually but good to record

File: src/components/subaccountTypeahead/SubaccountTypeahead.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      const { id, name } = item;
+
+      return {
+        ...getItemProps({ item, index }),
+        content: <Item name={name} id={id} />,
+        highlighted: highlightedIndex === index
+      };
+    });
+
+    const listClasses = cx(styles.List, {
```
Comment (jasonrhodes) [inline src/components/subaccountTypeahead/SubaccountTypeahead.js]: is this a common classnames convention?

Comment (tough-griff) [reply src/components/subaccountTypeahead/SubaccountTypeahead.js]: I've seen it before, and it's used in the docs for `classnames/bind` (see: https://github.com/JedWatson/classnames#alternate-bind-version-for-css-modules).

speaking of which, i'm gonna try using `bind` here.

File: src/components/reduxFormWrappers.js (hunk 3 above)
Comment (tough-griff) [reply src/components/reduxFormWrappers.js]: will do!

**Comment by tough-griff** [comment]
@jasonrhodes feedback addressed in 73ff4fb

**Comment by jasonrhodes** [comment]
L G T M

File: src/components/subaccountTypeahead/SubaccountTypeahead.module.scss [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  z-index: 1;
+
+  &.open {
+    opacity: 1;
+    pointer-events: auto;
+    transform: scale(1);
+  }
+}
+
+// TODO: Remove me once this is fixed in Matchbox
```
Comment (jonambas) [inline src/components/subaccountTypeahead/SubaccountTypeahead.module.scss]: Lets remove this for now, I'll make sure to get this fixed

Comment (tough-griff) [reply src/components/subaccountTypeahead/SubaccountTypeahead.module.scss]: 👍

**Pull request merged**